jax/numpy/lax_numpy.py

```diff
@@ -2618,6 +2618,20 @@ def polyadd(a, b):
   return lax.add(a,b)
 
 
+@_wraps(np.polysub)
+def polysub(a, b):
+  a = asarray(a)
+  b = asarray(b)
+
+  shape_diff = a.shape[0] - b.shape[0]
+  if shape_diff > 0:
+    b = concatenate((np.zeros(shape_diff, dtype=b.dtype), b))
+  else:
+    a = concatenate((np.zeros(shape_diff, dtype=b.dtype), a))
```

Comment:
This would be clearer and more efficient without constructing the temporary array by concatenation; something like this:
```
if b.shape[0] <= a.shape[0]:
  return a.at[-b.shape[0]:].sub(b)
else:
  return -b.at[-a.shape[0]:].sub(a)
```

Reply:
@jakevdp I tried this, but it gives me an error: `AttributeError: '_IndexUpdateRef' object has no attribute 'sub'`, hence I decided to go with the current approach.

Reply:
Ah, you're right. you have to use `add`. I think this would work
```
if b.shape[0] <= a.shape[0]:
  return a.at[-b.shape[0]:].add(-b)
else:
  return -b.at[-a.shape[0]:].add(-a)
```